src/main/java/org/killbill/billing/plugin/vertex/dao/VertexDao.java

```diff
@@ -108,18 +110,19 @@ public void addResponse(final UUID kbAccountId,
                                resultData.getDocumentDate() == null ? null : resultData.getDocumentDate().atStartOfDay(),
                                null,
                                resultData.getTotal() == null ? null : BigDecimal.valueOf(resultData.getTotal()),
-                              /* BigDecimal.valueOf(resultData.getDiscount().getDiscountValue()),
-                               BigDecimal.valueOf(resultData.totalExempt),
-                               BigDecimal.valueOf(resultData.totalTaxable),*/
+                               (resultData.getDiscount() == null || resultData.getDiscount().getDiscountValue() == null)
+                               ? null : BigDecimal.valueOf(resultData.getDiscount().getDiscountValue()),
+                               vertexResponseDataExtractor.getTotalTaxExempt(resultData.getLineItems()),
+                               vertexResponseDataExtractor.getTotalTaxable(resultData.getLineItems()),
                                resultData.getTotalTax() == null ? null : BigDecimal.valueOf(resultData.getTotalTax()),
-                               /* BigDecimal.valueOf(resultData.totalTaxCalculated),*/
+                               vertexResponseDataExtractor.getTotalTaxCalculated(resultData.getLineItems()),
```

Comment:
I see, we are using `resultData` at some places and `vertexResponseDataExtractor` at other to get relevant data. Can we let vertexResponseDataExtractor return all  required details?
Please see my other comment as its related to this.

Reply:
Reworked, thank you